src/main/java/com/google/api/codegen/discovery/MainDiscoveryProviderFactory.java

```diff
@@ -70,13 +72,19 @@ public static DiscoveryProvider defaultCreate(
           .build();
 
     } else if (id.equals(JAVA)) {
-      return CommonDiscoveryProvider.newBuilder()
-          .setContext(new JavaDiscoveryContext(service, apiaryConfig))
-          .setSnippetSetRunner(
-              new JavaSnippetSetRunner<Method>(SnippetSetRunner.SNIPPET_RESOURCE_ROOT))
-          .setSnippetFileName(id + "/" + DEFAULT_SNIPPET_FILE)
+      return ViewModelProvider.newBuilder()
+          .setApiaryConfig(apiaryConfig)
+          .setSnippetSetRunner(new CommonSnippetSetRunner(new CommonRenderingUtil()))
+          .setMethodToViewTransformer(new JavaMethodToViewTransformer())
+          .setOverrides(sampleConfigOverrides)
+          .setOutputRoot(
+              "autogenerated/"
+                  + apiaryConfig.getApiName()
+                  + "/"
+                  + apiaryConfig.getApiVersion()
+                  + "/"
+                  + service.getDocumentation().getOverview())
```

Comment:
It's a bit tricky, because of that `service.getDocumentation` bit. I'd like to leave it as is, until all 6 languages are complete. Then we can move it more easily.